custom_components/meural/pymeural.py

```diff
@@ -30,11 +34,15 @@ async def authenticate(
 
 
 class PyMeural:
-    def __init__(self, token, session: aiohttp.ClientSession):
-        self.token = token
+    def __init__(self, username, password, session: aiohttp.ClientSession):
+        self.username = username
+        self.password = password
         self.session = session
+        self.token = None
 
     async def request(self, method, path, data=None) -> Dict:
+        if self.token == None:
+            await self.get_new_token()
```

Comment:
Keep track if you fetched a token or not. 
```suggestion
        fetched_new_token = self.token is None
        if self.token == None:
            await self.get_new_token()
```